Add documentation about system-specific details (#92)
ukri-excalibur/excalibur-tests: SYSTEMS.md

You likely want to install this benchmarking framework outside of your home directory, for example inside `/work/<project code>/<project code>/${USER}`, where `<project code>` is your project code.

### Queue options

When submitting jobs to compute nodes, you need to specify the job queue and maybe the project account.
The former is specified by setting the [Quality of Service](https://docs.archer2.ac.uk/user-guide/scheduler/#quality-of-service-qos), the latter is necessary if your user account is associated to multiple projects on ARCHER2 and you need to [specify which one to use](https://docs.archer2.ac.uk/user-guide/scheduler/#specifying-resources-in-job-scripts) for the submitted jobs.
We cannot automatically set these options for you because they are user-specific, but when you run a benchmark you can use the `-J`/`--job-option` flag to `reframe` to add new job options.
Some examples:

```
reframe -c examples/sombrero -r --performance-report --system archer2 -J'--qos=serial'
reframe -c examples/sombrero -r --performance-report --system archer2 -J'--qos=short' -J'--account=t01'
```

> Note that jobs run on specific queues, like `serial` and `lowpriority`, do not charge the corresponding project, but you need to have at least 1 [Computing Resource (CU)](https://docs.archer2.ac.uk/user-guide/scheduler/#cus) in budget to access them.

Reviewer: Do we want to have this written down in our documentation? Perhaps I'd just rather link to the Archer2 docs.

Author: Fair enough
```suggestion
```

### Controlling CPU frequency

ARCHER2 allows [choosing the CPU frequency](https://docs.archer2.ac.uk/user-guide/energy/#controlling-cpu-frequency) during jobs by setting the environment variable `SLURM_CPU_FREQ_REQ` to specific values.
In ReFrame v3 the list of environment variables set by the framework is hold by the dictionary attribute called `variables`, and you can initialise it on the command line when running a benchmark with `-S`/`--setvar`.

> The syntax to set dictionary items is a comma-separated list of `key:value` pairs: `-S dict=key_1:value_1,key_2:value_2`

Author: You mean for example in https://github.com/ukri-excalibur/excalibur-tests/blob/c381d99227ce36fd458efaec3a52d001694a0ede/README.md#usage? :stuck_out_tongue: But that section doesn't explain how to supply multiple values, can move this bit of information there.

For example, to submit a benchmark using the lowest CPU frequency (1.5 GHz) you can use

```
reframe -c examples/sombrero -r --performance-report --system archer2 -J'--qos=serial' -S variables=SLURM_CPU_FREQ_REQ:1500000
```

## CSD3

### Queue options

When submitting jobs to compute nodes, you need to specify the job queue, with the `--account` option to the scheduler.
To do this, when you run a benchmark you can use the `-J`/`--job-option` flag to `reframe` to specify the account, for example:

```
reframe -c examples/sombrero -r --performance-report --system csd3-skylake:compute-node -J'--accout=<ACCOUNT>'
```

where `<ACCOUNT>` is the project you want to charge.
You can see the account balance of your projects with the `mybalance` command.

## DIaL3

### Queue options

When submitting jobs to compute nodes, you need to specify the job queue, with the `--account` option to the scheduler.
To do this, when you run a benchmark you can use the `-J`/`--job-option` flag to `reframe` to specify the account, for example:

```

> reframe -c examples/sombrero -r --performance-report --system csd3-skylake:compute-node -J'--accout=<ACCOUNT>'

Reviewer: ```suggestion
reframe -c examples/sombrero -r --performance-report --system dial3:compute-node -J'--accout=<ACCOUNT>'
```

```

> where `<ACCOUNT>` is the project you want to charge.

Author: It's true that there's some duplication here (as you could guess from my mistake in copying the name of the system in one of the examples), the account thing seems to be in common to the DiRAC systems, but it's not their exclusive (for example it's also on ARCHER2, but I think there it's relevant only if you have multiple projects attached to your account), and there may be other peculiarities specific to each system that we'll need to document anyway in a dedicated section (for example I have the feeling I forgot something about Tesseract or Tursa, but looking at my setup I can't tell what exactly)

### Queue options

When submitting jobs to compute nodes, you need to specify the job queue and maybe the project account.
The former is specified by setting the [Quality of Service](https://docs.archer2.ac.uk/user-guide/scheduler/#quality-of-service-qos), the latter is necessary if your user account is associated to multiple projects on ARCHER2 and you need to [specify which one to use](https://docs.archer2.ac.uk/user-guide/scheduler/#specifying-resources-in-job-scripts) for the submitted jobs.
We cannot automatically set these options for you because they are user-specific, but when you run a benchmark you can use the `-J`/`--job-option` flag to `reframe` to add new job options.
Some examples:

```
reframe -c examples/sombrero -r --performance-report --system archer2 -J'--qos=serial'
reframe -c examples/sombrero -r --performance-report --system archer2 -J'--qos=short' -J'--account=t01'
```

### Controlling CPU frequency

ARCHER2 allows [choosing the CPU frequency](https://docs.archer2.ac.uk/user-guide/energy/#controlling-cpu-frequency) during jobs by setting the environment variable `SLURM_CPU_FREQ_REQ` to specific values.

> In ReFrame v3 the list of environment variables set by the framework is hold by the dictionary attribute called `variables`, and you can initialise it on the command line when running a benchmark with `-S`/`--setvar`.

Reviewer: I think repetition is good, but I'd still point to the README here as well.

```suggestion
In ReFrame v3 the list of environment variables set by the framework is held by the dictionary attribute called `variables`, and you can initialise it on the command line when running a benchmark with `-S`/`--setvar`. For more details, see Setting environment variables in [`README.md`](./README.md).
```
